### ISSUE TITLE
Идея: визуализированный таймлайн по политическим делам

### ISSUE THREAD

**Issue by dkonasov** [developers-against-repressions/devs-against-the-machine#10]
Дел достаточно много и сложно их отслеживать. Было бы круто иметь некую базу, визуализированную в виде таймлайна с поиском по фио/делу/статье/группе статей. На таймлайне бы могли бы быть ключевые этапы дела - заседание по мере пресечения, апеляция, предварительные слушания и далее по тексту - с максимально возможной информацией по делу - онлайны, репортажи, документы и прочее. Очень бы помогло в информировании людей о политических репрессиях в России, как мне кажется.

**Comment by agricolamz** [comment]
Мне это интересно. Но я все делаю в R. Так как сейчас набегут всякие фронт- бэк- эндщики, я бы забился на работу по обдумыванию структуры базы, и автоматическому его пополнению, например, обкачиваю и предобробатываю данные откуда-нибудь.

**Comment by dkonasov** [comment]
> Мне это интересно. Но я все делаю в R. Так как сейчас набегут всякие фронт- бэк- эндщики, я бы забился на работу по обдумыванию структуры базы, ее представления, автоматическому ее пополнению, например, обкачиваю и предобробатываю данные откуда-нибудь.

О, это прям хорошее дело, да. А какую СУБД тебе удобнее будет использовать? Я пока, наверное буду считать себя фуллстеком, еще кто еще присоединится - буду рад. Фронт, думаю, стоит попробовать сделать на Vue, бэк - на ноде. Но это обсуждаемо, конечно.

**Comment by agricolamz** [comment]
А зачем что-то сложнее csv, json? В смысле дел много, но даже если включить все детали, вряд ли перевалит за тысячу строк.

Я это вижу вот [так](https://agricolamz.github.io/wwsd/), [исходник](https://github.com/agricolamz/wwsd), где на первой вкладке как раз визуализация таймлайна с зумом, и прочим интерактивом.

**Comment by KlonD90** [comment]
Было бы очень круто. Крутая визуализация процесса, чтобы было более очевидно fake'овость происходящего и легко ввести в курс дела.

**Comment by agricolamz** [comment]
Не знаю, стоит ли ставить какую-то цель типа "чтобы было очевидно x". Может быть стоит придерживаться какой-то отстраненности все таки и сообщать факты. Типа в 2010 было столько-то дел, которые мы согласно классификации специалиста X отнесли к политическим, а в 2019 столько-то.

**Comment by KlonD90** [comment]
> Не знаю, стоит ли ставить какую-то цель типа "чтобы было очевидно x". Может быть стоит придерживаться какой-то отстраненности все таки и пытаться сообщать факты. Типа в 2010 было столько-то дел, которые мы согласно классификации специалиста X отнесли к политическим, а в 2019 столько-то.

Ну тоже в целом неплохо но как это относится с таймлайнами?

**Comment by agricolamz** [comment]
Так, цель составить базу данных, а потом из нее можно делать любую визуализацию: таймлайн, что-то агрегированное по годам, что-то аггрегированное по какой-нибудь другой переменной...

**Comment by KlonD90** [comment]
> Так, цель составить базу данных, а потом из нее можно делать любую визуализацию: таймлайн, что-то агрегированное по годам, что-то аггрегированное по какой-нибудь другой переменной...

Crimestat основной. Политическими делами можно считать по базе данных мемориала

**Comment by KlonD90** [comment]
Однако если рассматривать все таки именно по политическим делам то можно кластеры публикаций касательно делать собирать и агрегировать их.

**Comment by agricolamz** [comment]
Положу-ка я [ссылку](https://memohrc.org/ru/bulletins/spisok-lic-priznannyh-politicheskimi-zaklyuchyonnymi-pravozashchitnym-centrom-memorial-i-1) здесь. Действительно в мемориале есть списки [1](https://memohrc.org/ru/tags/presleduyutsya-v-nastoyashchee-vremya) и [2](https://memohrc.org/ru/tags/sovremennye-politzaklyuchennye) c фильтрацией по региону. Наверное, наша цель представление в виде таймлайна с возможностью зума, плюс различные фильтры по параметрам, которые мы сможем докрутить и взять из других мест ([судебные акты](https://sudact.ru/), что еще?).

**Comment by dkonasov** [comment]
> А зачем что-то сложнее csv, json? В смысле дел много, но даже если включить все детали, вряд ли перевалит за тысячу строк. Но это мое мнение, наверное, это детали и обсуждаемо.
> 
> Я это вижу вот [так (нужно покликать на клавиши search, bibliography)](https://agricolamz.github.io/wwsd/), [исходник](https://github.com/agricolamz/wwsd), где на первой вкладке как раз визуализация таймлайна с зумом, и прочим интерактивом, например, как [здесь](https://github.com/daattali/timevis), и с агрегированными саммари-визуализациями по году, суду, судье и т. п.
> 
> Скорее важный вопрос, откуда брать данные? Т. е. что-то известно из судов и заключений, но вот откуда взять список именно политических дел? И где заканчивать? Только последнее время или идти до Даниэля Синявского? Декабристов?

Мне вот видится, что там должно быть несколько сущностей: событие, процесс, персоналии, материалы. Все это предполагает несложную, но все же реляционность, что приводит нас к необходимости полагаться на СУБД даже просто для удобства разработки.
С точки зрения внешнего вида мне это представляется как таймлайн с бусами-событиями, по кликам на каждое из которых можно увидеть список материалов к нему относящийся. Ну и мне кажется важной возможность отобразить несколько таймлайнов паралельно (хоть я и не могу сформулировать зачем это надо).

С контентом вопрос интересный, конечно. Я даже, наверное, не имею на этот счет хорошего мнения. Но это с другой стороны то, что можно обсудить после хакатона - на нем самом главное представить некий MVP, proof-of-concept с тестовыми данными, которые могут быть любыми, по идее (например, можно пока ограничится Московским делом).

**Comment by agricolamz** [comment]
> Мне вот видится, что там должно быть несколько сущностей: событие, процесс, персоналии, материалы. Все это предполагает несложную, но все же реляционность, что приводит нас к необходимости полагаться на СУБД даже просто для удобства разработки.

Думаю, это детали. Если бы я делал все на R, я бы успокоился на одной большой таблице. Если это не так, то пусть тот, кто будет делать, скажет, как ему удобнее.

> Ну и мне кажется важной возможность отобразить несколько таймлайнов паралельно (хоть я и не могу сформулировать зачем это надо).

Чтобы мочь группировать. Например группировка по полу, хоп, два таймлайна. Группировка по 

> например, можно пока ограничится Московским делом

Московское дело, достаточно раскручено, я бы взял что-нибудь еще нераскрученное, чтобы мочь сравнивать что мы може делать.

**Comment by dkonasov** [comment]
> Думаю, это детали. Если бы я делал все на R, я бы успокоился на одной большой таблице. Если это не так, то пусть тот, кто будет делать, скажет, как ему удобнее.

Надо, наверное, уже где-то прикинуть архитектуру этого счастья и "сверить часы". Тот вариант, который у меня в голове слабо умещается в одну таблицу. Впрочем, я слабо представляю как это делается в мире R.

> Чтобы мочь группировать. Например группировка по полу, хоп, два таймлайна.

Хм, а зачем с практической точки зрения может быть нужна группировка по полу? Мне видится тут основным юзкейсом необходимость сопоставить ход двух дел (возможно с целью установить некую связь с ними).





> Московское дело, достаточно раскручено, я бы взял что-нибудь еще вдовесок нераскрученное, чтобы можно было сравнить, что мы можем/хотим делать.

И правда. Ну и еще нам наверное понадобится как минимум еще одно дело для демонстрации возможности фильтрации по делам. То есть получается стоит взять одно большое (Московское) и одно поменьше (из текущих дел мне навскидку на ум приходит дело Дмитриева).

**Comment by agricolamz** [comment]
> Надо, наверное, уже где-то прикинуть архитектуру этого счастья и "сверить часы".

Какие переменнные: дело, ФИО, возраст, пол, место рождения, место проживания, специальность?, работа?, дата задержания, ОВД, срок пребывание в ОВД, срок пребывания в ..., дата суда 1, дата суда 2, ..., судья 1, судья 2...  адвокат 1, адвокат 2... прокурор 1, прокурор 2... требования обвинения 1, требования обвинения 2, ... решение судьи 1, решение судьи 2,  дата освобождения, Что еще? Пока это укладывается в одну таблицу :) Правда я бы единицей базы данных делал дату, тогда будуте сразу удобнее жить, но из той таблицы, которую я описал, можно все трансформировать в таблицу где каждая дата --- новая строчка.

> Хм, а зачем с практической точки зрения может быть нужна группировка по полу.

Чтобы показать, что женщин не задерживали, нужна эта переменная. Точно так же как если мы хотим показать, что не задерживают пожилых людей. Чем больше переменных -- тем больше люди смогут увидеть. Мне кажется, не стоит ограничивать себя набором посылов, часто люди могут увидеть в данных что-то, что мы не закладывали.

> И правда. Ну и еще нам наверное понадобится как минимум еще одно дело для демонстрации возможности фильтрации по делам. То есть получается стоит взять одно большое (Московское) и одно поменьше (из текущих дел мне навскидку на ум приходит дело Дмитриева).

Я бы брал из соображений, сбор данных для чего легко автоматизируются

**Comment by agricolamz** [comment]
Ну типа [вот](https://agricolamz.github.io/2019.12.13-15_dev_against_the_machine/timeline_prototype.html) я за час стяпал. [Исходник](https://github.com/agricolamz/2019.12.13-15_dev_against_the_machine).

Вот я вижу, что у нас сбоку есть панелька, на которую мы кликаем и он показывает разбивку по разным переменным.

**Comment by agricolamz** [comment]
А еще я вот думаю, что было бы интересно получить мнение какого-нибудь юриста из ОВД инфо и т. п., чтобы подсказать, какие переменные могут быть интересны, откуда вытаскивать данные и т. п., я, наверное, даже могу попробовать спросить знакомую.

**Comment by agricolamz** [comment]
Обновил [пример](https://agricolamz.github.io/2019.12.13-15_dev_against_the_machine/timeline_prototype.html), теперь там больше разбивок.

**Comment by dkonasov** [comment]
Пример выглядит симпатично. Но я бы, наверное, в качестве нод на таймлайне отрисовывал бы события, а не персоналий. Плюс хотелось бы, чтобы по клику открывались ссылки (или даже превьюхи) на репортаж с события и карточки привязанных к нему персоналий (обвиняемые, судьи, прокуроры и. т. д.)

**Comment by agricolamz** [comment]
Так, туда внутрь можно пихать любой html/js -- это вопрос визуализации.
Тогда нужно поменять структуру базы: строчкой должно быть тогда событие.

Данные: https://politpressing.org/, ОВД Инфо писали, что готовы сотрудничать. @loalexandra

**Comment by agricolamz** [comment]
А еще мне пришло в голову, что можно визуализировать разными способами:

* таймлайн
* ежегодными плашками, как на гитхабе
* в виде календаря (с экспортом в google и ios календарь)

**Comment by agricolamz** [comment]
Я поговорил с <https://github.com/behlbor>, и они вроде согласны дать нам данные, они вроде в json'ах в MongoDB.

Нужно определиться с командой. Кто нам нужен?

**Comment by agricolamz** [comment]
В общем, обсуждение мертво, если что я создал команду 42.

**Comment by agricolamz** [comment]
https://github.com/agricolamz/2019.12.13-15_dev_against_the_machine/tree/master

Здесь результаты.
